File: app/src/main/java/com/song/deviceinfo/utils/ApplicationUtils.java
package com.song.deviceinfo.utils;

import android.content.Context;
import android.content.pm.PackageInfo;

/**
 * Created by chensongsong on 2021/9/28.
 */
public class ApplicationUtils {

    public static boolean isPkgInstalled(Context context, String pkgName) {
        PackageInfo packageInfo;
        try {
            packageInfo = context.getPackageManager().getPackageInfo(pkgName, 0);
        } catch (Exception e) {
            packageInfo = null;
        }
        return packageInfo != null;
    }

}

File: app/src/main/java/com/song/deviceinfo/utils/ApplicationUtils.kt
package com.song.deviceinfo.utils

import android.content.Context

/**
 * Created by chensongsong on 2021/9/28.
 */
object ApplicationUtils {
    @JvmStatic
    fun isPkgInstalled(context: Context, pkgName: String): Boolean {
        var packageInfo = try {
            context.packageManager.getPackageInfo(pkgName, 0)
        } catch (e: Exception) {
            null
        }
        return packageInfo != null
    }
}
